packages/syft/src/syft/service/network/network_service.py

```diff
@@ -586,20 +542,13 @@ def add_route_on_peer(
         """
         # creates a client on the remote node based on the credentials
         # of the current node's client
-        remote_client = peer.client_with_context(context=context)
-        if remote_client.is_err():
-            return SyftError(
-                message=f"Failed to create remote client for peer: "
-                f"{peer.id}. Error: {remote_client.err()}"
-            )
-        remote_client = remote_client.ok()
+        remote_client = peer.client_with_context(context=context).unwrap()
```

Comment:
add previous error message
                message=f"Failed to create remote client for peer: "
                f"{peer.id}. Error: {remote_client.err()}"